dev-java/protobuf-java/protobuf-java-3.20.1-r1.ebuild

```diff
@@ -0,0 +1,90 @@
+	inherit git-r3
+
+	EGIT_REPO_URI="https://github.com/protocolbuffers/protobuf.git"
+	EGIT_SUBMODULES=()
+	EGIT_CHECKOUT_DIR="${WORKDIR}/${PARENT_P}"
+else
+	SRC_URI="
+		https://github.com/protocolbuffers/protobuf/archive/v${PARENT_PV}.tar.gz
+			-> ${PARENT_P}.tar.gz
+	"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86 ~amd64-linux ~x86-linux ~x64-macos"
+fi
+
+DESCRIPTION="Google's Protocol Buffers - Java bindings"
+HOMEPAGE="
+	https://developers.google.com/protocol-buffers/
+"
+
+LICENSE="BSD"
+SLOT="0/31"
+
+S="${WORKDIR}/${PARENT_P}/java"
+
+BDEPEND="
+	dev-libs/protobuf:${SLOT}
+"
+DEPEND="
+	>=virtual/jdk-1.8:*
+"
+RDEPEND="
+	>=virtual/jre-1.8:*
+"
+
+JAVA_SRC_DIR="core/src/main/java"
+JAVA_RESOURCE_DIRS="core/src/main/resources"
+JAVA_AUTOMATIC_MODULE_NAME="com.google.protobuf"
+JAVA_JAR_FILENAME="protobuf.jar"
+
+PATCHES=(
+)
+
+src_prepare() {
+	pushd "${WORKDIR}/${PARENT_P}" > /dev/null || die
+	default
+	popd > /dev/null || die
+
+	# Remove bundled jars
+	java-pkg_clean
+
+	java-pkg-2_src_prepare
+
+	# Copy resources from ../src/google/protobuf according to
+	# https://github.com/protocolbuffers/protobuf/blob/v3.20.1/java/core/pom.xml#L45-L61
+	mkdir -p core/src/main/resources/google/protobuf/compiler || die
+	cp {../src,core/src/main/resources}/google/protobuf/compiler/plugin.proto || die
+	cp ../src/google/protobuf/{any,api,descriptor,duration,empty,field_mask,source_context,struct,timestamp,type,wrappers}.proto \
+		"${JAVA_RESOURCE_DIRS}/google/protobuf" || die
+
+	# Generate 146 .java files according to
+	# https://github.com/protocolbuffers/protobuf/blob/v3.20.1/java/core/generate-sources-build.xml
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/any.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/api.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/compiler/plugin.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/descriptor.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/duration.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/empty.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/field_mask.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/source_context.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/struct.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/timestamp.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/type.proto || die
+	"${BROOT}/usr/bin/protoc" --java_out=core/src/main/java -I../src ../src/google/protobuf/wrappers.proto || die
```

Comment:
Those lines could be condensed, see https://github.com/gentoo/gentoo/compare/master...vaukai:gentoo:protobuf-java.
At least when getting  to preparing the tests you won't be wiling to translate [all these lines](https://github.com/protocolbuffers/protobuf/blob/v3.21.6/java/core/generate-test-sources-build.xml#L7-L52) into the ebuild :-)

Reply:
Thanks for the suggestion, updated.